src/Stack/Ghci.hs

```diff
@@ -747,14 +751,15 @@ wantedPackageComponents _ _ _ = S.empty
 
 checkForIssues :: HasLogFunc env => [GhciPkgInfo] -> RIO env ()
 checkForIssues pkgs = do
-    unless (null issues) $ borderedWarning $ do
-        logWarn "Warning: There are cabal settings for this project which may prevent GHCi from loading your code properly."
-        logWarn "In some cases it can also load some projects which would otherwise fail to build."
-        logWarn ""
-        mapM_ (logWarn . RIO.display) $ intercalate [""] issues
-        logWarn ""
-        logWarn "To resolve, remove the flag(s) from the cabal file(s) and instead put them at the top of the haskell files."
-        logWarn ""
+    when (length pkgs > 1) $ borderedWarning $ do
```

Comment:
This looks a bit confusing to me. The code reads as "only show the issues if there are multiple packages." I think we know that there are only issues if there are multiple packages, but it's not obvious at this part of the code. I'd recommend one of the following:

* Comment explaining this
* Add an assertion like `assert (length pkgs > 1 || null issues)`
* Only wrapping the 'It isn't yet possible" comment below with `when (length pkgs > 1)`

Reply:
You can't have 2 packages with different flags when you have only 1 - does this sound as a good explanation?
But it makes sense to make it more explicit that `issues` in this case are about conflicting flags between different packages.

Reply:
Yes, that's a good explanation. I'm just thinking that with a generic term like `issues`, it's easy to read it as "all of the potential issues with a package, not just the ones that could arise from conflicting flags or different CWDs."

Reply:
Exactly and even different CWDs don't go into these `issues`.

Reply:
I've pushed an update addressing this